client/post-editor/post-editor.jsx

```diff
@@ -213,6 +215,18 @@ export const PostEditor = React.createClass( {
 		}
 	},
 
+	getSelectedText: function() {
+		const selectedText = tinyMce.activeEditor.selection.getContent() || null;
+		if ( this.state.selectedText !== selectedText ) {
+			this.setState( { selectedText: selectedText || null } );
+		}
+	},
+
+	onEditorKeyUp: function() {
+		this.getSelectedText();
```

Comment:
This should be a `debounce()`d call - recounting with every key up could have noticeable performance issues, particularly with large selections or on slower devices.